src/core/track.cpp

```diff
@@ -155,14 +155,17 @@ int FFMS_Track::FrameFromPTS(int64_t PTS) const {
     F.PTS = PTS;
 
     auto Pos = std::lower_bound(begin(), end(), F, PTSComparison);
+    while (Pos != end() && Pos->Hidden && Pos->PTS <= PTS)
```

Comment:
Pos->PTS will never be less than PTS (lower_bound finds the first greater than or equal), so this check is a little suspicious. Assuming this is merely redundant and not wrong, this check can be shoved in PTSComparison as `std::tie(FI1.PTS, FI1.Hidden) < std::tie(FI2.PTS, FI2.Hidden)` rather than being a separate loop that's duplicated.